shark_turbine/kernel/wave/promotion.py

```diff
@@ -38,3 +40,15 @@ def promote_node(node: Read | Write, address_space: IndexSymbol):
         )
         allocate_node.add_to_graph(node.graph)
         apply_promotion_pattern(node, allocate_node)
+
+
+def promote_placeholders(graph: CapturedTrace):
+    for node in graph.get_root_graph().nodes:
+        custom = get_custom(node)
+        if isinstance(custom, Read) or isinstance(custom, Write):
+            if not custom.type:
+                continue
+            idxc = IndexingContext.current()
+            address_space = custom.type.address_space.subs(idxc.subs)
```

Comment:
For my own education, is type for custom.type.address_space symbolic expression? if so, why can't we just have it be directly `index_symbol("$SHARED_ADDRESS_SPACE")`?

Reply:
Looking at the test I see that we are setting it as another symbol `ADDRESS_SPACE_K` where K is argument number. Is this to make the kernels more modification friendly S.T we can use same kernel but change up TestLaunchContext to modify address space?

Reply:
Makes sense, thanks!